### ISSUE TITLE
'Tensor' object has no attribute 'numpy'

### ISSUE THREAD

**Issue by tu1258** [tensorflow/tensorflow#43464]
I'll just post this as bug issue because if I post other issues you guys would just tell me to post at stackoverflow every time while stackoverflow never helps and I actually tried many methods they had posted before I open this issue. 

So I try to print the output of each layer and got this error. `tf.executing_eagerly` and `model.run_eagerly` are all TRUE. I've checked most issues on the github and this situation is rare so those just don't help. Here is the code(very simple). 
[model_print_data.txt](https://github.com/tensorflow/tensorflow/files/5263474/model_print_data.txt)
This may provide some details. Much appreciated.

**Comment by bhack** [comment]
Before we start to investigate this we need a very, very minimal standalone example to reproduce your issue that we could just copy, paste and run or a colab (if possible without filesystem dependencies as we prefer dummy inputs).

**Comment by tu1258** [comment]
Thanks! I'll attach the files. Thanks so much for not ignoring me again. This is supposed to be run on bash with venv. 
[test.zip](https://github.com/tensorflow/tensorflow/files/5263587/test.zip)
**The step to install enviroment**

```
pip install virtualenv
virtualenv -p python3 ./venv
source ./venv/bin/activate
pip install --upgrade pip
pip install tensorflow==2.3.0
pip install pydot
pip install pydotplus
sudo apt-get install graphviz
pip install graphviz
pip install frozendict
pip install numpy
pip install absl-py
```

Just run 
```
python -m kws_test.train.model_print_data \
--data_dir ./data_test \
--train_dir ./kws_print \
--window_size_ms 40.0 \
--window_stride_ms 20.0 \
--mel_num_bins 80 \
--dct_num_features 30 \
--resample 0.15 \
--time_shift_ms 100 \
--train 0 \
--debug 1 \
--generate 0 \
--print 1 \
--kernel_size '(3,1)' \
--first_kernel '(1,1)' \
--channels '36, 36, 36, 36, 36, 36, 36' \
--groups 1 \
--mobile 0 \
--residual 0 \
--dropout 0.0 \
&> ./kws_print.txt

```
at the `test` folder and you can reproduce 100% the same.

**Comment by bhack** [comment]
@tu1258 I think it is too articulated. You really need **to minimize** the code surface to let use to reproduce your problem. If not we need to route you on the Stackoverflow channel.

**Comment by tu1258** [comment]
Oh, let me explain a bit. 
The code is only in `kws_test` folder. 
`kws_print` folder is the "saved_model" folder.
`data_test` only contains 1 raw audio. 😄 
Since there are a few custom layers(all are in `feature` folder) and they are needed when doing inference. (Of cource there are some unnecessary files, I can remove them.) However, the most important is `model_print_data` which is highly readable. It just reads 1 audio (yes_data.csv) and run `model.predict` and only import the model(tcn.py) and base_parser.py. If you find that it's truly necessary to shrink the code, I'll do it in few days tho. But please take a look first. Much appreciated.

**Comment by tu1258** [comment]
Hi, @bhack I minimized code. There is only `model_print_data.py`. model folder and `best_weights` are being loaded to predict. `yes_data.csv` is the test data.
[test.zip](https://github.com/tensorflow/tensorflow/files/5265791/test.zip)
you just run 
```
python model_print_data.py \
&> ./kws_print.txt

```

**Comment by ravikyram** [comment]
@tu1258 

Please, see similar issue #27519 and [SO link](https://stackoverflow.com/questions/52357542/attributeerror-tensor-object-has-no-attribute-numpy) and see if it helps you.

Can you try with  `tf.compat.v1.disable_eager_execution() ` ans see if the issue still persists. Thanks!

**Comment by tu1258** [comment]
@ravikyram 

```
Traceback (most recent call last):
  File "/home/training/anaconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/training/anaconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/training/r08943133/kws_test/train/model_print_data.py", line 96, in <module>
    tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/tensorflow/python/platform/app.py", line 40, in run
    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "/home/training/r08943133/kws_test/train/model_print_data.py", line 38, in main
    print_test_data(flags)
  File "/home/training/r08943133/kws_test/train/model_print_data.py", line 72, in print_test_data
    predictions = model.predict(test_fingerprints)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/tensorflow/python/keras/engine/training_v1.py", line 992, in predict
    use_multiprocessing=use_multiprocessing)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/tensorflow/python/keras/engine/training_arrays.py", line 714, in predict
    callbacks=callbacks)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/tensorflow/python/keras/engine/training_arrays.py", line 386, in model_iteration
    batch_outs = f(ins_batch)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/tensorflow/python/keras/backend.py", line 3825, in __call__
    run_metadata=self.run_metadata)
  File "/home/training/r08943133/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1472, in __call__
    run_metadata_ptr)
tensorflow.python.framework.errors_impl.InvalidArgumentError: 2 root error(s) found.
  (0) Invalid argument: Input to reshape is a tensor with 16000 values, but the requested shape has 2048000
	 [[{{node feature_layer/data_frame/frame/Reshape}}]]
  (1) Invalid argument: Input to reshape is a tensor with 16000 values, but the requested shape has 2048000
	 [[{{node feature_layer/data_frame/frame/Reshape}}]]
	 [[reshape_1/_187]]
0 successful operations.
0 derived errors ignored.
```

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.3 and was able to reproduce the issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/12c94eb10b1e53716332933e222a4647/untitled385.ipynb).Thanks!

**Comment by bhack** [comment]
@tu1258 You code is not correct as you cannot retrieve the output layer in this way.
If you need to get layers outputs please use something like https://keras.io/getting_started/faq/#how-can-i-obtain-the-output-of-an-intermediate-layer-feature-extraction

**Comment by tu1258** [comment]
At first, thanks for the both reply. @bhack I've tried the method you mention above (you could actually see the comment line just  below those failed print code) and it still fails.

Noted that those commented code are not exactly what I run. I've tried with this and got error `TypeError: call() got an unexpected keyword argument 'outputs'`
```
intermediate_layer_model = model(inputs=model.input,
                                   outputs=model.get_layer(index=0).output)
  intermediate_output = intermediate_layer_model(data)
  print(intermediate_output)
```
Much appreciated!

**Comment by bhack** [comment]
@tu1258 Can you close this ticket now?

Also about your initial claim
> I'll just post this as bug issue because if I post other issues you guys would just tell me to post at stackoverflow every time while stackoverflow never helps and I actually tried many methods they had posted before I open this issue.

I think that it is hard also for stackoverflow if you open questions with a not very very minimized code/use case. 
Generally you to need to investigate on your side and minimize your code cause most of the time you can reproduce/redouce your problem with very few lines of code (as in this case you have still too much code distraction in your zip). Any other additional line generate noise and it requires more time for Stackoverflow contributors or ticket triaging so you will have lower probability to recieve support.

**Comment by tu1258** [comment]
I'll close it. 

> I think that it is hard also for stackoverflow if you open questions with a not very very minimized code/use case.

To be honest, I didn't plan to post code in a regular issue. This is the first time since I think there are already lots of same issue and I still couldn't solve it after reading them. (it turns out that I seem to have some misunderstand though, I apologize) I understand the pain of looking into code written by others. That's why I only attach that `model_print_data.py` because I didn't expect you guys to reproduce the situation. I only expect to get some advice. And if we look back from now, the issue can be solved based on what I attached if I explained it a bit. I am sorry that I didn't explain my code is just an "load model" thing.

About stackoverflow, I think many issue (that are recommended to post on stackoverflow, usually usage problem) are because of lack of document. Not saying tf are bad, because building complete document is impossible. We all know those are not bug report or feature request and not improving the tensorflow framework. But what can they do what if stackoverflow can't solve problems? Users can only come here for help. I believe people who are willing to spend time opening an issue have already searched for answer on google/stackoverflow.
